src/ASM/ASMs2D.C

```diff
@@ -1907,7 +1909,7 @@ bool ASMs2D::integrate (Integrand& integrand,
 #endif
 
           // Cartesian coordinates of current integration point
-          X = Xnod * fe.N;
+          X.assign(Xnod * fe.N);
```

Comment:
In this method, where the integration point parameters are provided through an array, the only change needed is to add `X.u = elmPts[ip].data()` (like I did for the equivalent ASMu2D method, but forgot for ASMs2D), and we can skip the param temporary and the assign-method.

Reply:
done.